### PR TITLE
FORTECH-8: Add TooltipDecorator

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Feature Added
Created Agate TooltipDecorator component.

Comments
Added new TooltipDecorator component in Agate. Created styles for Gallium and Silicon skins, added unit tests and integrated component in StoryBook, added tooltip as Button prop.

Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### CODE REVIEW

File: styles/variables-base.less [hunk 1]
```diff
@@ -540,3 +540,15 @@
...
+// ---------------------------------------
+@agate-tooltip-label-font-size: 	inherit;
+@agate-tooltip-offset: 				9px;
+@agate-tooltip-point-width: 		12px;
+@agate-tooltip-point-height: 		12px;
+@agate-tooltip-border-width: 		1px;
+@agate-tooltip-border-radius:		3px;
+@agate-tooltip-label-height:		45px;
+@agate-tooltip-label-font-weight:	@agate-button-font-weight;
+@agate-tooltip-padding: 			3px 18px;
```
Comment (daniel-stoian-lgp) [inline styles/variables-base.less:554]: alignment looks broken

File: TooltipDecorator/Tooltip.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
+import kind from '@enact/core/kind';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Skinnable from '../Skinnable';
+import TooltipLabel from './TooltipLabel';
```
Comment (ybsung) [inline TooltipDecorator/Tooltip.js:6]: We prefer to add the one empty line between the imported module out of the current directory and the imported module in the current directory.

```
import Skinnable from '../Skinnable';

import TooltipLabel from './TooltipLabel';
```

**Comment by ybsung** [comment]
I don't know why, but there are some lint errors which are not found in the `develop` branch. It might be resolved after refreshing your branch based on the `develop` branch.

```
Parsing: [#######             ] (84/232) agate/RangePicker
Parse error: Unknown content '`aria-valuetext`' in raw/agate/RangePicker

Parse error: Unknown content '`aria-valuetext`' in raw/agate/RangePicker:undefined
Parsing: [#######             ] (86/232) agate/Picker
Parse error: Unknown content '`aria-valuetext`' in raw/agate/Picker

Parse error: Unknown content '`aria-valuetext`' in raw/agate/Picker:undefined
Parsing: [########            ] (90/232) agate/ThumbnailItem
Parse error: unknown tag @agate in raw/agate/ThumbnailItem:6
```

File: styles/variables-base.less [hunk 3]
```diff
@@ -540,3 +540,15 @@
 @agate-togglebutton-indicator-height: 6px;
 @agate-togglebutton-indicator-transform: none;
 @agate-togglebutton-indicator-width: auto;
+
+// Tooltip
+// ---------------------------------------
+@agate-tooltip-label-font-size: 	inherit;
+@agate-tooltip-offset: 				9px;
```
Comment (ybsung) [inline styles/variables-base.less:547]: Please check the indentation here.

File: styles/variables-base.less (hunk 1 above)
Comment (ybsung) [inline styles/variables-base.less:554]: Same above.

File: TooltipDecorator/TooltipLabel.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import kind from '@enact/core/kind';
+import {isRtlText} from '@enact/i18n/util';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import css from './Tooltip.module.less';
+
+/**
+ * {@link agate/TooltipDecorator.TooltipLabel} is a stateless tooltip component with
+ * Moonston styling applied.
```
Comment (daniel-stoian-lgp) [inline TooltipDecorator/TooltipLabel.js:10]: agate

**Comment by ybsung** [comment]
There is no JSDoc parse error. All the errors were fixed.

File: TooltipDecorator/Tooltip.module.less [hunk 5]
```diff
@@ -0,0 +1,474 @@
...
+			height: ((@agate-tooltip-point-height / 2) + 2px);
+			top: 50%;
+
+			&::after {
+				bottom: 99%;
+				left: 0;
+			}
+		}
+
+		.tooltipLabel {
```
Comment (ybsung) [inline TooltipDecorator/Tooltip.module.less:70]: `.moon-small-button-text();` was removed while adapting the moonstone TooltipDecorator to the Agate. Was it intended?

Comment (stanca-pop-lgp) [reply TooltipDecorator/Tooltip.module.less:70]: Yes, I did not find that necessary.

File: Button/Button.js [hunk 6]
```diff
@@ -264,13 +265,15 @@ const ButtonBase = kind({
...
  * @memberof agate/Button
+ * @mixes agate/TooltipDecorator.TooltipDecorator
  * @mixes ui/Button.ButtonDecorator
  * @mixes spotlight/Spottable.Spottable
  * @mixes agate/Skinnable.Skinnable
  * @public
  */
 const ButtonDecorator = compose(
 	Pure,
+	TooltipDecorator({tooltipDestinationProp: 'decoration'}),
```
Comment (ybsung) [inline Button/Button.js:276]: Is there any reason to apply `TooltipDecorator` by default in `Button`? It could make the `Button` heavy to render.

**Comment by ybsung** [comment]
It seems that `TooltipDecorator` component is from `moonstone`. Only the JSDocs are updated.

File: Button/Button.js (hunk 6 above)
Comment (stanca-pop-lgp) [reply Button/Button.js:276]: I will remove it and only use it where needed. Thank you!

**Comment by ybsung** [comment]
Now everything looks good to me.

**Pull request merged**